src/component.jl

```diff
@@ -192,48 +199,57 @@ function simulate_component(rng, c::MixedModelComponent, design::AbstractDesign)
 
 
     # empty epoch data
-    epoch_data_component = zeros(Int(length(c.basis)), length(design))
+    #epoch_data_component = zeros(Int(length(c.basis)), length(design))
 
     # residual variance for lmm
     σ_lmm = 0.0001
-    if 1 == 1
-        named_random_effects = weight_σs(c.σs, 1.0, σ_lmm)
-        θ = createθ(m; named_random_effects...)
-        simulate!(deepcopy(rng), m.y, m; β = c.β, σ = σ_lmm, θ = θ)
-
-        # save data to array
-        #@show size(m.y)
-        #@show size(c.basis)
 
+    named_random_effects = weight_σs(c.σs, 1.0, σ_lmm)
+    θ = createθ(m; named_random_effects...)
+    @debug named_random_effects, θ, m.θ
+    try
+        simulate!(deepcopy(rng), m.y, m; β = c.β, σ = σ_lmm, θ = θ)
+    catch e
+        if isa(e, DimensionMismatch)
+            @warn "Most likely your σs's do not match the formula!"
+        elseif isa(e, ArgumentError)
+            @warn "Most likely your β's do not match the formula!"
+        end
+        rethrow(e)
+    end
 
-        epoch_data_component = kron(c.basis, m.y')
+    # save data to array
 
 
-    else
```

Comment:
For which case does the `else`-statement exist?

Reply:
it used to exist for the code in comment now - it is in case you want to weight betas and sigmas differently (e.g. different basis function for each.

it is _much_ slower though